.NET/Microsoft.Recognizers.Text.DateTime/Parsers/CJK/BaseCJKDatePeriodParser.cs

```diff
@@ -0,0 +1,1622 @@
+            var pr2 = this.config.DateParser.Parse(er[1], referenceDate);
+
+            if (er.Count >= 2)
+            {
+                // @TODO Refactor code to remove the cycle between BaseDatePeriodParser and its config.
+                var dateContext = GetYearContext(er[0].Text, er[1].Text, text);
+
+                if (pr1.Value == null || pr2.Value == null)
+                {
+                    return ret;
+                }
+
+                pr1 = dateContext.ProcessDateEntityParsingResult(pr1);
+                pr2 = dateContext.ProcessDateEntityParsingResult(pr2);
+
+                // When the case has no specified year, we should sync the future/past year due to invalid date Feb 29th.
+                if (dateContext.IsEmpty() && (DateContext.IsFeb29th((DateObject)((DateTimeResolutionResult)pr1.Value).FutureValue)
+                                              || DateContext.IsFeb29th((DateObject)((DateTimeResolutionResult)pr2.Value).FutureValue)))
+                {
+                    (pr1, pr2) = dateContext.SyncYear(pr1, pr2);
+                }
+            }
+
+            DateObject futureBegin = (DateObject)((DateTimeResolutionResult)pr1.Value).FutureValue,
+                futureEnd = (DateObject)((DateTimeResolutionResult)pr2.Value).FutureValue;
+            DateObject pastBegin = (DateObject)((DateTimeResolutionResult)pr1.Value).PastValue,
+                pastEnd = (DateObject)((DateTimeResolutionResult)pr2.Value).PastValue;
+
+            if (futureBegin > futureEnd)
+            {
+                futureBegin = pastBegin;
+            }
+
+            if (pastEnd < pastBegin)
+            {
+                pastEnd = futureEnd;
+            }
+
+            ret.Timex = TimexUtility.GenerateDatePeriodTimex(futureBegin, futureEnd, DatePeriodTimexType.ByDay, pr1.TimexStr, pr2.TimexStr);
+
+            if (pr1.TimexStr.StartsWith(Constants.TimexFuzzyYear) && futureBegin.CompareTo(DateObject.MinValue.SafeCreateFromValue(futureBegin.Year, 2, 28)) <= 0 && futureEnd.CompareTo(DateObject.MinValue.SafeCreateFromValue(futureBegin.Year, 3, 1)) >= 0)
+            {
+                // Handle cases like "2月28日到3月1日".
+                // There may be different timexes for FutureValue and PastValue due to the different validity of Feb 29th.
+                ret.Comment = Constants.Comment_DoubleTimex;
+                var pastTimex = TimexUtility.GenerateDatePeriodTimex(pastBegin, pastEnd, DatePeriodTimexType.ByDay, pr1.TimexStr, pr2.TimexStr);
+                ret.Timex = TimexUtility.MergeTimexAlternatives(ret.Timex, pastTimex);
+            }
+
+            ret.FutureValue = new Tuple<DateObject, DateObject>(futureBegin, futureEnd);
+            ret.PastValue = new Tuple<DateObject, DateObject>(pastBegin, pastEnd);
+            ret.Success = true;
+
+            return ret;
+        }
+
+        // handle like "前两年" "前三个月"
+        private DateTimeResolutionResult ParseNumberWithUnit(string text, DateObject referenceDate)
```

Comment:
Same.

Reply:
this method exists only in CJK implementation.